### PR TITLE
Fixed #30335, #29139 -- Fixed crash when ordering or aggregating over a nested JSONField key transforms.

### PR DESCRIPTION
**Pull request by cansarigol**
Ticket: [#30335](https://code.djangoproject.com/ticket/30335)

### CODE REVIEW

File: tests/postgres_tests/test_json.py [hunk 1]
```diff
@@ -162,6 +162,33 @@ def test_ordering_by_transform(self):
...
+                    {'title': 'Dames en heren'}
+                ]
+            }),
+            JSONModel.objects.create(field={
+                'translations': [
+                    {'title': 'Apples and pears'},
+                    {'title': 'Appels en peren'}
+                ]
+            }),
+        ]
```
Comment (felixxm) [inline tests/postgres_tests/test_json.py]: IMO we don't need to create new objects (`objs = [...]`), we should be able to change queries and reuse existing ones, e.g.
```python
def test_ordering_grouping_by_key_transform(self):
    self.assertSequenceEqual(
        JSONModel.objects.filter(field__d__0__isnull=False).order_by('field__d__0'),
        [self.objs[8]],
    )
    qs = JSONModel.objects.filter(field__isnull=False).values('field__d__0').annotate(
        count=Count('field__d__0'),
    ).order_by('count')
    self.assertQuerysetEqual(qs, [1, 10], operator.itemgetter('count'))
```

**Comment by cansarigol** [comment]
hi, @felixxm thanks. changed test and added note.

**Comment by cansarigol** [comment]
Have I should added other related tickets (29139, 28762, 30085) into doc as a note?

**Comment by felixxm** [comment]
30085 is a closed duplicate so definitely no.

For 29139 and 28762 I will try later to push tests that will confirm that.

File: docs/releases/2.2.1.txt [hunk 2]
```diff
@@ -33,3 +33,6 @@ Bugfixes
 * Reverted an optimization in Django 2.2 (:ticket:`29725`) that caused the
   inconsistent behavior of ``count()`` and ``exists()`` on a reverse
   many-to-many relationship with a custom manager (:ticket:`30325`).
+
+* Fixed crash when ``ordering`` or ``aggregating`` over a deep
```
Comment (charettes) [inline docs/releases/2.2.1.txt]: Both `ordering` and `aggregating` shouldn't be in backtips and _deep_ should probably be replaced by _nested_.

**Pull request merged**